jxl/src/frame/modular/transforms.rs

```diff
@@ -398,6 +317,32 @@ pub struct TransformStepChunk {
     incomplete_deps: usize,
 }
 
+impl TransformStepChunk {
+    // Mark that one dependency of this transform is ready, and potentially runs the transform,
+    // returning the new buffers that are now ready.
+    pub fn dep_ready(
+        &mut self,
+        buffer_info: &mut [ModularBufferInfo],
+    ) -> Result<Vec<(usize, usize)>> {
+        self.incomplete_deps = self.incomplete_deps.checked_sub(1).unwrap();
```

Comment:
will this unwrap panic sometimes, or is it certain that `self.incomplete_deps` is never `0`?

Reply:
It should never be 0 unless there's a logic bug somewhere.